pyproject.toml

```diff
@@ -55,50 +54,36 @@ developer =[
     "black[jupyter]==22.*",
     "flake8",
     "bandit[toml]", # code scanning
-    "tox",          # testing
     "hydra-core",   # configuration
 ]
 
 engine = [
-    "adversarial-robustness-toolbox==1.12.2",
-    "Pillow",         # Data dependencies
-    "boto3",          # Needed for armory.data.utils
-    "botocore",       # Needed for armory.data.utils
-    "ffmpeg-python",  # Needed for armory.utils.export
-    "opencv-python",  # Needed for CARLA baseline scenario
-    "pydub",          # this is in ART's extra-requires
-    "tensorboardx",
-    "tidecv",         # Needed for TIDE metrics
-]
-
-datasets = [
-    "tensorflow-datasets >= 4.6.0",
-    "tensorflow >= 2.10.0",
-    "protobuf",
-]
-
-datasets-builder = [
-    "armory-testbed[datasets,engine]",
-    "apache-beam >= 2.22.0",
-]
-
-math = [
-    "numpy",
-    "scipy",
-    "scikit-learn < 1.1.0",
-    "matplotlib",
+    "adversarial-robustness-toolbox == 1.12.2",
+    "Pillow",                # Data dependencies
+    "boto3 == 1.24.89",      # Needed for armory.data.utils
+    "botocore == 1.27.89" ,  # Needed for armory.data.utils
+    "ffmpeg-python",         # Needed for armory.utils.export
+    "pydub",               # this is in ART's extra-requires
+    "tidecv",              # Needed for TIDE metrics
+    # Both `opencv-python` and `opencv-python-headless` must specify
+    # the same version.
+    "opencv-python == 4.5.5.62",          # Needed for CARLA baseline scenario
+    "opencv-python-headless == 4.5.5.62", # Needed for CARLA baseline scenario
+    "tensorboardx == 2.5.1",
 ]
 
 pytorch = [
     "armory-testbed[engine,datasets,math]",
     "torch",
-    "torchvision",
-    "torchaudio",
+    "torchaudio == 0.12.1",
+    "torchvision == 0.13.1",
```

Comment:
Why are these specified? If they are specified, `torch` version should be specified as well.